packages/flutter_tools/test/commands.shard/hermetic/test_test.dart

```diff
@@ -804,6 +804,40 @@ dev_dependencies:
       ProcessManager: () => FakeProcessManager.any(),
     });
   });
+
+  group('File Reporter', () {
+    testUsingContext('defaults to unset null value', () async {
+      final FakeFlutterTestRunner testRunner = FakeFlutterTestRunner(0);
+
+      final TestCommand testCommand = TestCommand(testRunner: testRunner);
+      final CommandRunner<void> commandRunner = createTestCommandRunner(testCommand);
+
+      await commandRunner.run(const <String>[
+        'test',
+        '--no-pub',
+      ]);
+      expect(testRunner.lastFileReporterValue, null);
+    }, overrides: <Type, Generator>{
+      FileSystem: () => fs,
+      ProcessManager: () => FakeProcessManager.any(),
+    });
+    testUsingContext('when set --file-reporter value is passed on', () async {
+      final FakeFlutterTestRunner testRunner = FakeFlutterTestRunner(0);
+
+      final TestCommand testCommand = TestCommand(testRunner: testRunner);
+      final CommandRunner<void> commandRunner = createTestCommandRunner(testCommand);
+
+      await commandRunner.run(const <String>[
+        'test',
+        '--no-pub',
+        '--file-reporter=json:out.jsonl'
+      ]);
+      expect(testRunner.lastFileReporterValue, 'json:out.jsonl');
```

Comment:
Can we also add an expect that something was written to out.jsonl

Reply:
Do you want a test that checks if anything was written or do you want to check the specific output?

I feel the later is a little brittle, as it depends on the output of the `dart test` tool.

Reply:
@Jasguerrero I may be wrong, but while exploring the code to implement this request I noted that in the tests we actually don't run the test command, as we use a `FakeFlutterTestRunner` instead of a regular `_FlutterTestRunnerImpl`.

Testing this case would required an integration test. Is this what you mean?

Reply:
> Do you want a test that checks if anything was written or do you want to check the specific output?

It was more like a sanity check that the file was created with something in it regardless the content (as you mentioned `dart test` should handle the content)

> Testing this case would required an integration test. Is this what you mean?

Good point. Looking at the code you could leverage https://github.com/flutter/flutter/blob/3.3/packages/flutter_tools/test/integration.shard/test_test.dart#L167 the problem with those tests is that it uses the local file system (then asserting on the stdout), writing a file to it is not the best practice. For now I think the current tests in this PR are good enough.